hyper/http20/stream.py

```diff
@@ -277,16 +279,19 @@ def close(self, error_code=None):
         """
         # FIXME: I think this is overbroad, but for now it's probably ok.
         if not (self.remote_closed and self.local_closed):
-            try:
-                self._conn.reset_stream(self.stream_id, error_code or 0)
-            except h2.exceptions.ProtocolError:
-                # If for any reason we can't reset the stream, just tolerate
-                # it.
-                pass
-            else:
-                self._send_cb(
-                    self._conn.data_to_send(), tolerate_peer_gone=True
-                )
+            with self._conn as conn:
+                send = False
+
+                try:
+                    conn.reset_stream(self.stream_id, error_code or 0)
+                except h2.exceptions.ProtocolError:
+                    # If for any reason we can't reset the stream, just
+                    # tolerate it.
+                    pass
+                else:
+                    send = True
+            if send:
+                self._send_outstanding_data(tolerate_peer_gone=True)
```

Comment:
You may as well just bring this directly into the `else`, rather than test the conditional.